### PR TITLE
Update database

### PR DESCRIPTION
**Pull request by DanWilkins2107**

### CODE REVIEW

**Comment by dwulfse** [comment]
is this meant to be a PR? its just all the availability stuff again

**Comment by DanWilkins2107** [comment]
Yeah reset still doesn't work. Idk why plz let me know if u can figure out why, all changes from availability have been made on this PR

File: components/Availability/AvailabilityCalendar.jsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            availabilityInfo[year][month][day] &&
+            availabilityInfo[year][month][day][hour]
+        ) {
+            return availabilityInfo[year][month][day][hour];
+        }
+        return "none";
+    };
+
+    useEffect(() => {
+        scrollViewRef.current.scrollTo({ y:635, animated: true });
```
Comment (dwulfse) [inline components/Availability/AvailabilityCalendar.jsx]: this is currently scrolling back up to 8:00 every time I change availability

Comment (dwulfse) [reply components/Availability/AvailabilityCalendar.jsx]: also doesn't scroll to 8:00 when opening the availability screen

File: components/Availability/AvailabilityCalendar.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    });
+    return (
+        <View className="h-full">
+            <View className="h-20 flex-row justify-center items-center">
+                <Text className="font-semibold text-lg">Choose Date:</Text>
+                <DateTimePicker
```
Comment (dwulfse) [inline components/Availability/AvailabilityCalendar.jsx]: bit of a weird one but this is formatted as "6 Jun 2024" 99% of the time then goes to "30/06/2024" for some dates, very randomly

File: screens/ProductionScreens/Availability.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            <Text className="text-3xl font-extrabold">Set Availability</Text>
+            <View className="w-full h-5/6 border-b mb-8">
+                <AvailabilityCalendar
+                    availabilityInfo={availabilityInfo}
+                    setAvailabilityInfo={setAvailabilityInfo}
+                />
+            </View>
+            <View className="flex flex-row justify-around w-full">
+                <SmallFormButton title="Submit" backgroundColor="bg-green-400" onPress={onSubmit} />
+                <SmallFormButton title="Reset" backgroundColor="bg-white" onPress={onReset} />
```
Comment (dwulfse) [inline screens/ProductionScreens/Availability.jsx:85]: can't work out why reset isn't working yet. works the first time then never again

**Review by dwulfse — CHANGES_REQUESTED** [review]

**Comment by dwulfse** [comment]
this is so cool omg

File: components/Availability/AvailabilityCalendar.jsx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                        setDate(editDate(date, 7));
+                    }}
+                >
+                    <Icon name="arrow-circle-right" size={25} />
+                </TouchableOpacity>
+            </View>
+
+            <View className="flex-row w-full h-14 justify-around border-b ">
+                <View className="w-16 border-r" />
+                <View className="flex-row flex-1 border-t border-r bg-slate-100">
```
Comment (dwulfse) [inline components/Availability/AvailabilityCalendar.jsx]: try no border

File: components/Availability/AvailabilityCalendar.jsx [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                                                )}
+                                            </Pressable>
+                                        );
+                                    })}
+                                </View>
+                            );
+                        })}
+                    </View>
+                </View>
+                {/* Any Thoughts on whether we want this line when the week is not current? */}
```
Comment (dwulfse) [inline components/Availability/AvailabilityCalendar.jsx]: quite undecided, reckon leave it everywhere for now just cos its cool, then can think about it later

File: screens/ProductionScreens/Availability.jsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+export default function Availability({ navigation, route }) {
+    const [availabilityInfo, setAvailabilityInfo] = useState({});
+    const [initialAvailabilityInfo, setInitialAvailabilityInfo] = useState({});
+    const db = firebase_db;
+    const auth = firebase_auth;
+    const playCode = route.params.productionCode;
+    const { setAlert } = useContext(AlertContext);
+
+    const onSubmit = async () => {
```
Comment (dwulfse) [inline screens/ProductionScreens/Availability.jsx:20]: disable buttons on submit/ reset

File: components/Availability/AvailabilityCalendar.jsx [hunk 7]
```diff
@@ -0,0 +1,244 @@
+import { View, Text, ScrollView, Pressable, TouchableOpacity } from "react-native";
+import { useState, useRef, useCallback } from "react";
+import DateCircles from "./DateCircles";
+import Icon from "react-native-vector-icons/FontAwesome5";
+import { set } from "firebase/database";
```
Comment (dwulfse) [inline components/Availability/AvailabilityCalendar.jsx]: set not used

**Review by dwulfse — CHANGES_REQUESTED** [review]

File: components/Availability/AvailabilityCalendar.jsx [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+                        setDate(editDate(date, 7));
+                    }}
+                >
+                    <Icon name="arrow-circle-right" size={25} />
+                </TouchableOpacity>
+            </View>
+
+            <View className="flex-row w-full h-14 justify-around border-b ">
+                <View className="w-16" />
+                <View className="flex-row flex-1  bg-slate-100">
```
Comment (dwulfse) [inline components/Availability/AvailabilityCalendar.jsx]: ugly background

**Review by dwulfse — CHANGES_REQUESTED** [review]
just performance to worry about

**Review by dwulfse — APPROVED** [review]

**Pull request merged**